library/md.c

```diff
@@ -248,7 +248,24 @@ static int md_can_use_psa(const mbedtls_md_info_t *info)
         return 0;
     }
 
+    /* psa_can_do_hash() below is used to determine if the PSA Crypto core has
+     * been initialized or not. However it is a vendor specific function which
+     * is not part of the standard PSA Crypto interface so we cannot make any
+     * assumption about it being supported in the PSA Crypto provider. In this
+     * case we try to simply abort a freshly created hash operation. If it fails
+     * we assume that either psa_crypto_init() has not been called or the PSA
+     * Crypto provider does not support hash.
+     */
+#if defined(MBEDTLS_PSA_CRYPTO_C)
     return psa_can_do_hash(alg);
```

Comment:
Shouldn't we call `psa_can_do_hash` in the client-only case as well? I don't see why we wouldn't.

Reply:
As discussed during the meeting `md_can_use_psa` in case of `MBEDTLS_PSA_CRYPTO_CLIENT && !MBEDTLS_PSA_CRYPTO_C` now:

- checks that the hash algorithm is supported in PSA through `psa_alg_of_md`
- checks that `psa_crypto_init` was called through the fake aborting of a freshly initialized hash operation.

Is that reasonably OK?